projects/cds/src/cms-components/merchandising/carousel/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -26,13 +26,20 @@ export class MerchandisingCarouselComponent {
   title$: Observable<string> = this.componentData$.pipe(
     map(data => data.title)
   );
+  numberToDisplay$: Observable<string> = this.componentData$.pipe(
+    map(data => data.numberToDisplay)
+  );
 
   items$: Observable<Observable<Product>[]> = this.componentData$.pipe(
-    map(data => data.strategy),
-    distinctUntilChanged(),
-    switchMap(strategyId =>
-      this.cdsMerchandisingProductService.loadProductsForStrategy(strategyId)
+    distinctUntilChanged(
+      (previous, current) => previous.strategy === current.strategy
```

Comment:
Let's drop this, it's wasn't the best idea of mine :)

Reply:
I found the distinctUntilKeyChanged('strategy') instead